### ISSUE TITLE
Provide more context for DoesNotExist Exception

### ISSUE THREAD

**Issue by spacemanspiff2007** [tortoise/tortoise-orm#741]
With FastAPI it's possible to register a central exception handler for certain errors.
It's elegant to register a listener for ``DoesNotExist``, because then the error handling doesn't have to be created in every route.
However I'd like to give a proper error message what kind of Model did not exist and with what primary key it was queried.
Currently that is not possible because ``DoesNotExist`` doesn't provide the required context.

Ideal implementation:

````python
from tortoise.exceptions import DoesNotExist

 try:
    obj = await MyModel.get(id=1)
except DoesNotExist as e:
    assert e.model is MyModel
    assert e.pk_name == 'id'
    assert e.pk_value == 1
````

PS: It would be nice if we could rename ``DoesNotExist`` to ``DoesNotExistError``

**Comment by marcoaaguiar** [comment]
Hi, spacemanspiff2007, I agree with you. More information is always better than less information (at least for errors) :)
Just keep in mind that `Model.get` works not only queries by pk fields but for any query that returns a single result in the query set.

**Comment by spacemanspiff2007** [comment]
> Just keep in mind that ``Model.get`` works not only queries by pk fields but for any query that returns a single result in the query set.

You are absolutely right! Maybe we can find a way to attach this information to the error or maybe we can define a shortcut to query a Model by primary key, e.g.
```python
obj = await MyModel[1]
```

**Comment by marcoaaguiar** [comment]
Regarding your first point, do you mean attaching the query information? ({"id":1},)

**Comment by spacemanspiff2007** [comment]
yes, exactly

**Comment by marcoaaguiar** [comment]
Regarding your second point, it should be implemented already (haven't checked tho) #386
